components/ModEditFacilityForm.vue

```diff
@@ -0,0 +1,290 @@
+        />
+        <ModInputField
+            v-model="submissionFormFields.mapLatitude.value"
+            data-testid="submission-form-mapLatitude"
+            :label="$t('modSubmissionForm.labelFacilityMapLatitude')"
+            type="text"
+            :placeholder="$t('modSubmissionForm.placeholderTextFacilityMapLatitude')"
+            :required="true"
+            :input-validation-check="validateFloat"
+            :invalid-input-error-message="$t('modSubmissionForm.inputErrorMessageFacilityMapLatitude')"
+        />
+        <ModInputField
+            v-model="submissionFormFields.mapLongitude.value"
+            data-testid="submission-form-mapLongitude"
+            :label="$t('modSubmissionForm.labelFacilityMapLongitude')"
+            type="text"
+            :placeholder="$t('modSubmissionForm.placeholderTextFacilityMapLongitude')"
+            :required="true"
+            :input-validation-check="validateFloat"
+            :invalid-input-error-message="$t('modSubmissionForm.inputErrorMessageFacilityMapLongitude')"
+        />
+    </div>
+    <ModHealthcareProfessionalSearchbar data-testid="submission-form-doctor-search" />
+    <button
+        v-show="screenStore.activeScreen === ModerationScreen.EditFacility"
+        type="submit"
+        class="bg-currentColor text-white font-bold py-2 px-4 my-2 rounded w-56"
+    >
+        {{ $t('modSubmissionForm.updateButtonText') }}
+    </button>
+</template>
+
+<script lang="ts" setup>
+import { type Ref, ref } from 'vue'
+import { useModerationScreenStore, ModerationScreen } from '~/stores/moderationScreenStore'
+import { validateAddressLineEn,
+    validateAddressLineJa,
+    validateNameEn,
+    validateNameJa,
+    validatePhoneNumber,
+    validateCityEn,
+    validateEmail,
+    validateFloat,
+    validatePostalCode,
+    validateWebsite,
+    validateCityJa } from '~/utils/formValidations'
+
+const screenStore = useModerationScreenStore()
+
+const submissionFormFields = {
+    // contactFields
+    nameEn: ref('') as Ref<string>,
+    nameJa: ref('') as Ref<string>,
+    phone: ref('') as Ref<string>,
+    website: ref('') as Ref<string>,
+    email: ref('') as Ref<string>,
+    // addressesFields
+    postalCode: ref('') as Ref<string>,
+    prefectureEn: ref('') as Ref<string>,
+    cityEn: ref('') as Ref<string>,
+    addressLine1En: ref('') as Ref<string>,
+    addressLine2En: ref('') as Ref<string>,
+    prefectureJa: ref('') as Ref<string>,
+    cityJa: ref('') as Ref<string>,
+    addressLine1Ja: ref('') as Ref<string>,
+    addressLine2Ja: ref('') as Ref<string>,
+    // googleMapsFields
+    googlemapsURL: ref('') as Ref<string>,
+    mapLatitude: ref('') as Ref<string>,
+    mapLongitude: ref('') as Ref<string>
+} as { [key: string]: Ref }
+
+const listPrefectureJapanEn: Ref<string[]> = ref([
+    'Hokkaido', 'Aomori', 'Iwate', 'Miyagi', 'Akita',
+    'Yamagata', 'Fukushima', 'Ibaraki', 'Tochigi', 'Gumma', 'Saitama', 'Chiba', 'Tokyo', 'Kanagawa',
```

Comment:
Perhaps we should move these to the store since they're being used in multiple components. It would be more inline with "DRY" principles.

Reply:
Makes sense to me